app/assets/javascripts/admin/views/statistic/project_progress_report_view.js

```diff
@@ -1,42 +1,100 @@
 // @flow
 import * as React from "react";
-import { Spin, Table, Card } from "antd";
+import { Icon, Spin, Table, Card } from "antd";
 import Utils from "libs/utils";
+import FormatUtils from "libs/format_utils";
 import { getProjectProgressReport } from "admin/admin_rest_api";
 import type { APIProjectProgressReportType } from "admin/api_flow_types";
 import TeamSelectionForm from "./team_selection_form";
 
-const { Column } = Table;
+const { Column, ColumnGroup } = Table;
+
+const RELOAD_INTERVAL = 10 * 60 * 1000; // 10 min
 
 type State = {
+  settingsVisible: boolean,
+  teamId: ?string,
   data: Array<APIProjectProgressReportType>,
   isLoading: boolean,
+  updatedAt: ?number,
 };
 
 class ProjectProgressReportView extends React.PureComponent<{}, State> {
   state = {
+    settingsVisible: true,
     data: [],
+    teamId: undefined,
     isLoading: false,
+    updatedAt: null,
   };
 
-  async fetchData(teamId: ?string) {
+  componentDidMount() {
+    this.componentIsMounted = true;
+    this.autoRefresh();
+  }
+
+  componentWillUnmount() {
+    this.componentIsMounted = false;
+  }
+
+  componentIsMounted: boolean = false;
+
+  async fetchData(suppressLoadingState?: boolean = false) {
+    const { teamId } = this.state;
     if (teamId == null) {
       this.setState({ data: [] });
     } else {
-      this.setState({ isLoading: true });
+      if (!suppressLoadingState) {
+        this.setState({ isLoading: true });
+      }
       const progessData = await getProjectProgressReport(teamId);
-      this.setState({ data: progessData, isLoading: false });
+      this.setState({ data: progessData, updatedAt: Date.now() });
+      if (!suppressLoadingState) {
+        this.setState({ isLoading: false });
+      }
     }
   }
 
+  /* eslint-disable no-await-in-loop */
+  async autoRefresh() {
+    await Utils.sleep(RELOAD_INTERVAL);
+    while (this.componentIsMounted) {
```

Comment:
Isn't this a rather verbose re-implementation of `setInterval`?

Reply:
The only reason you need to save `componentIsMounted` is because you couldn't otherwise end the loop. This has also already been solved with `clearInterval`